comments/board.py

```diff
@@ -0,0 +1,441 @@
+# Game_Piece Super & Sub class
+class Game_Piece():
+    def __init__(self, color, number):
+        self.name = color + "_piece_" + str(number)
+        self.xy_coord = None
+        self.team = color
+
+    def update_position(self, xy_coord):
+        self.xy_coord = xy_coord
+
+
+class Checkers_Game_Piece(Game_Piece):
+    def __init__(self, color, number):
+        super().__init__(color, number)
+        self.is_king = False
+        self.moves = {
+            "move_nw": None,
+            "move_ne": None,
+            "move_sw": None,
+            "move_se": None
+        }
```

Comment:
One thing I don't like about this is that there is no indication about what the values of this dictionary is supposed to be. And you have this check_valid_moves() function that resets the values of this dictionary. So check_valid_moves is a misleading title. set_moves() would make more sense because you're actually setting the dictionary values there.